src/app/meshes/views/MeshView.vue

```diff
@@ -32,29 +32,35 @@
 import { KTabs } from '@kong/kongponents'
 import { useRouter } from 'vue-router'
 
+import { useMeshRoutes, useI18n } from '@/utilities'
+const { t } = useI18n()
+
 const router = useRouter()
-const items = [
-  {
-    hash: 'mesh-detail-view',
-    title: 'Overview',
-  },
-  {
-    hash: 'services-list-view',
-    title: 'Services',
-  },
-  {
-    hash: 'gateways-list-view',
-    title: 'Gateways',
-  },
-  {
-    hash: 'data-planes-list-view',
-    title: 'Data Plane Proxies',
-  },
-  {
-    hash: 'policies',
-    title: 'Policies',
-  },
-]
+
+// dig down into where our mesh tabs are and automatically
+// convert the routes to tabs
+const meshRoutes = useMeshRoutes()
+  .find(item => item.name === 'mesh-index-view')?.children
+  ?.find(item => item.name === 'mesh-detail-view')?.children
+  ?.find(item => item.name === 'mesh-abstract-view')?.children ?? []
```

Comment:
**Suggestion**: This really seems quite brittle and reliant to a _very_ specific route nesting and naming.

Wouldn’t something like this be more robust?

```ts
const meshRoutes = router.getRoutes().find((route) => route.name === 'mesh-abstract-view') ?? []
```

Reply:
Oh definitely yes, exactly the kind of thing I was after! ty!

Reply:
(As an aside, we should probably start putting route names in constants in the routes config file and export them and never use string literals for them.)